src/lib/Libwin/passwd.c

```diff
@@ -4028,13 +4035,17 @@ setuser_close_handle()
 	if (setuser_hdle != INVALID_HANDLE_VALUE) {
 		CloseHandle(setuser_hdle);
 
-		/* look in internal array for saved usertoken handle */
-		for (i=0; i<user_num; i++) {
-			pwdp = &user_array[i];
+		if(user_num == 0){
+			CLEAR_HEAD(passwd_cache_ll);
+		}
 
+		/* look in internal array for saved usertoken handle */
+		pwdp = (struct passwd*) GET_NEXT(passwd_cache_ll);
```

Comment:
convert into for loop (all places) like this:
	for (pwdp = (struct passwd*) GET_NEXT(passwd_cache_ll); pwdp;	pwdp = (struct passwd*) GET_NEXT(pwdp->pw_allpasswds)) {